src/main/scala/mesosphere/marathon/MarathonScheduler.scala

```diff
@@ -148,22 +149,24 @@ class MarathonScheduler(
     }
   }
 
-  protected def updateHomeRegion(masterInfo: MasterInfo): Unit = {
```

Comment:
I would save `MasterInfo` completely providing helper methods like `getHomeRegion` to access parts of it.